views/js/qtiCommonRenderer/renderers/interactions/AssociateInteraction.js

```diff
@@ -355,38 +351,189 @@ define([
+                $activeChoice = $target;
+                $activeChoice.addClass('active');
 
-                    $activeChoice = $(this);
-                    $activeChoice.addClass('active');
+                $resultArea.find('>li>.target').filter(function(){
+                    return $(this).data('serial') !== targetSerial;
+                }).addClass('empty');
 
-                    $resultArea.find('>li>.target').filter(function(){
-                        return $(this).data('serial') !== serial;
-                    }).addClass('empty');
+                $choiceArea.find('>li:not(.deactivated)').filter(function(){
+                    return $(this).data('serial') !== targetSerial;
+                }).addClass('empty');
+            };
 
-                    $choiceArea.find('>li:not(.deactivated)').filter(function(){
-                        return $(this).data('serial') !== serial;
-                    }).addClass('empty');
 
-                    //append trash bin:
-                    var $bin = $('<span>', {'class' : 'icon-undo remove-choice', 'title' : __('remove')});
-                    $bin.on('mousedown', function(e){
-                        e.stopPropagation();
-                        _unsetChoice($activeChoice);
-                        _resetSelection();
-                    });
-                    $(this).append($bin);
-                }
+            // Point & click handlers
+
+            interact($container.selector).on('tap', function() {
+                _resetSelection();
+            });
+
+            interact($choiceArea.selector + ' >li').on('tap', function(e) {
+                var $target = $(e.currentTarget);
+                e.stopPropagation();
+                _handleChoiceActivate($target);
+                e.preventDefault();
+            });
+
+            interact($resultArea.selector + ' >li>div').on('tap', function(e) {
+                var $target = $(e.currentTarget);
+                e.stopPropagation();
+                _handleResultActivate($target);
+                e.preventDefault();
+            });
 
+            interact(binSelector).on('tap', function (e) {
+                e.stopPropagation();
+                _unsetChoice($activeChoice);
+                _resetSelection();
+                e.preventDefault();
             });
 
             if(!interaction.responseMappingMode){
                 _setInstructions(interaction);
             }
 
+
+            // Drag & drop handlers
+
+            if (self.getOption && self.getOption("enableDragAndDrop") && self.getOption("enableDragAndDrop").associate) {
+                isDragAndDropEnabled = self.getOption("enableDragAndDrop").associate;
+            }
+
+            // Chrome/Safari ugly fix: manually drop element when the mouse leaves the item runner iframe
+            // should be removed when the old test runner is discarded
+            function _iFrameDragFix(draggableSelector, target) {
```

Comment:
Would be a good idea to move all the _iFrameDragFix() and _iFrameDragFixOff() functions into the interact helper.
If variations are needed, could add an optional callback or fire event